packages/lesswrong/components/common/CloudinaryImage2.tsx

```diff
@@ -20,46 +23,71 @@ function cloudinaryPropsToStr(props) {
   return sb.join(",");
 }
 
+function makeCloudinaryImageUrl (publicId: string, cloudinaryProps: CloudinaryPropsType) {
+  return `https://res.cloudinary.com/${cloudinaryCloudNameSetting.get()}/image/upload/${cloudinaryPropsToStr(cloudinaryProps)}/${publicId}`
+}
+
 // Cloudinary image without using cloudinary-react. Allows SSR. See:
 // https://github.com/LessWrong2/Lesswrong2/pull/937 "Drop cloudinary react"
 // https://github.com/LessWrong2/Lesswrong2/pull/964 "Temporarily revert removal of cloudinary-react"
-const CloudinaryImage2 = ({width, height, objectFit, publicId, imgProps, className}: {
+const CloudinaryImage2 = ({width, height, objectFit, publicId, imgProps, header, className}: {
```

Comment:
If I'm understanding correctly, setting `header` does the following things:
1. Fetches an image which is twice as high as the user originally requested from cloudinary (the image is still displayed at the requested height though)
2. The fetched image is fetched at the initial width and then displayed at 100% width (except the code to fetch it at initial width is almost never going to be hit)
3. The image receives a srcSet

I feel confused about these things:
1. Why is this specific to header images? I think your comment is saying that cloudinary by default provides us with too low of resolution of images, which may be true, but why is it only true for header images? I would think if anything it is less important for header images to be high resolution.
2. Overriding the user provided width seems confusing to me. I would at least put a comment that `width` is ignored if `header`. Better would be to create an ADT which enforces this, though maybe that's overkill. Also I'm not sure why the initial width is the correct thing to use here – this means we will download a different resolution of image based on what was uploaded to cloudinary, right? I feel like that's unexpected/will result in us downloading images at too high of a resolution sometimes.
3. Again I feel confused about why this is specific to headers. Maybe instead the flag should be `highResolutionImage` or something?

Reply:
First finding: I should have spent 15 minutes googling to see whether high-dpi displays automatically requested the higher resolution images. They do. So we don't need to manually add that part in, and burn resolution when it's unnecessary.

Reply:
Your understanding:

1. ~As mentioned in my previous comment, I've now changed this, as this behavior is done automatically by the browser.~ Oh, nevermind, I just brought it back, in the height dimension. High DPI clients will select the larger width, but we don't have a way to discriminate between a truly wide low DPI client and a narrower but high DPI client. The former would ideally get an image of height `height`, but the latter of height `2*height`. I'm going to make the call that we should bias towards the high-DPI side.
2. ~You say "almost never", which is directionally true, but I think you mean it like "IE support", which is not true. Image source is simply the fallback if it's larger than the largest size in your srcset. 3000 is pretty wide, and captures the largest laptop screens, but desktop users with wide monitors will still see the `iw` images.~ Nope, I'm just wrong here. Teach me to make assumptions.
3. Yep.

Your questions:

1. Why is this specific to full-width-header images: It's not that cloudinary provides us with too *low* resolution images, rather they're too *high*, and take too long to download. We have the opportunity to use srcsets to improve the load times substantially, especially on mobile.
  Why does this matter more for header images? Because for other images, you're usually able to provide some sensible heights and widths, which mean you're requesting the image of the roughly correct size already. Why *can't* you do this for header images? Because they vary so. freaking. much. in size and shape, you really want to get differently shaped images based on the screensize. One way to do this is to get the full image and let the browser crop it, but it saves much loading speed to request different srcsets.
  Ideally we could request srcsets for other images, but I think it would look like: `        ${makeCloudinaryImageUrl(publicId, {...cloudinaryProps, ...{w: 2*width, h: 2*height}})} 2x`.
2.You may have noticed during our call that I was thinking about this. The other problem, besides the one inherent to destructuring in typescript, is that in the process of registering the component, we do some fancy work to Omit some special props from the required args when it is used by the consumer. As a result of our fanciness, we actually destroy the union information. The long and the short of this is, an ADT would not work here.
  On the topic of requesting too-large images, I didn't realize that if we requested too large of an image from Cloudinary, that they would upres it on the server and waste bandwidth sending it over the wire. It makes sense. If we request an image of a certain size, that's all they know. They don't know that I'm constraining the width and height of the image element directly. Fortunately, we can set a relatively low maximum size for which we request a specific resolution, and just request `iw` above that. 
3. I think I should change it to fullWidthHeader, which I've done.